contracts/lib/time_series/convert.cairo

```diff
@@ -0,0 +1,90 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.bool import TRUE, FALSE
+from starkware.cairo.common.pow import pow
+from starkware.cairo.common.math_cmp import is_le
+from starkware.cairo.common.math import unsigned_div_rem, assert_not_zero
+
+func convert_via_usd{range_check_ptr}(a_price_in_usd, b_price_in_usd, b_decimals) -> felt {
+    let (pow_) = pow(10, b_decimals);
+    let (output_, _) = unsigned_div_rem(a_price_in_usd * pow_, b_price_in_usd);
```

Comment:
Overflow check for the multiplication?

Reply:
it would raise an exception in that case.  If we think rationally, we have 10 ** 18 * 10 ** 18 with 10 ** 75 space for this operation.  So realistically we would not hit this limit.  In terms of an overflow check, we could confirm a_price_in_usd and pow_ are both under some common sense value like 10 ** 32